community/io/src/main/java/org/neo4j/io/pagecache/impl/muninn/MuninnPageCursor.java

```diff
@@ -265,10 +299,33 @@ protected void assertPagedFileStillMapped()
 
     // --- IO methods:
 
+    private void checkBounds( int position )
+    {
+        if ( position > size )
+        {
+            throw new IndexOutOfBoundsException( getOutOfBoundsMessage( position ) );
+        }
+    }
+
+    private String getOutOfBoundsMessage( int position )
+    {
+        if ( size > 0 )
+        {
+            return "Position " + position + " is greater than the upper " +
+                   "page size bound of " + size;
+        }
+        else
+        {
+            return "The PageCursor is not bound to a page. " +
+                   "Maybe next() returned false or was not called, or the cursor has been closed.";
+        }
+    }
+
     @Override
     public final byte getByte()
     {
-        byte b = page.getByte( offset );
+        checkBounds( offset + 1 );
```

Comment:
Okay, I didn't know they had those constants. I still think they are poorly named. Now that I know what they are, I think they make no difference to readability. Apparently `javac` inlines them, so they also make no difference to the bytecode.